model_api/python/model_api/models/visual_prompting.py

```diff
@@ -0,0 +1,675 @@
+                )
+            else:
+                used_idx = self._used_indices
+        else:
+            reference_feats, used_idx = reference_features
+
+        original_shape = np.array(image.shape[:2])
+        image_embeddings = self.encoder_model(image)
+
+        total_points_scores, total_bg_coords = _get_prompt_candidates(
+            image_embeddings=image_embeddings,
+            reference_feats=reference_feats,
+            used_indices=used_idx,
+            original_shape=original_shape,
+            threshold=self._threshold,
+            num_bg_points=self._num_bg_points,
+            default_threshold_target=self._default_threshold_target,
+            image_size=self._image_size,
+            downsizing=self._downsizing,
+        )
+
+        predicted_masks: defaultdict[int, list] = defaultdict(list)
+        used_points: defaultdict[int, list] = defaultdict(list)
+        for label in total_points_scores:
+            points_scores = total_points_scores[label]
+            bg_coords = total_bg_coords[label]
+            for points_score in points_scores:
+                if points_score[-1] in [-1.0, 0.0]:
+                    continue
+
+                x, y = points_score[:2]
+                is_done = False
+                for pm in predicted_masks.get(label, []):
+                    # check if that point is already assigned
+                    if pm[int(y), int(x)] > 0:
+                        is_done = True
+                        break
+                if is_done:
+                    continue
+
+                point_coords = np.concatenate(
+                    (np.array([[x, y]]), bg_coords), axis=0, dtype=np.float32
+                )
+                point_coords = self.decoder_model.apply_coords(
+                    point_coords, original_shape
+                )
+                point_labels = np.array([1] + [0] * len(bg_coords), dtype=np.float32)
+                inputs_decoder = {
+                    "point_coords": point_coords[None],
+                    "point_labels": point_labels[None],
+                    "orig_size": original_shape[None],
+                }
+                inputs_decoder["image_embeddings"] = image_embeddings
+
+                prediction = self._predict_masks(
+                    inputs_decoder, original_shape, True
+                )
+                prediction.update({"scores": points_score[-1]})
+
+                predicted_masks[label].append(
+                    prediction[self.decoder_model.output_blob_name]
+                )
+                used_points[label].append(points_score)
+
+        # check overlapping area between different label masks
+        _inspect_overlapping_areas(predicted_masks, used_points)
+
+        return (predicted_masks, used_points)
+
+    def reset_reference_info(self) -> None:
```

Comment:
Since there is a property called `reference_features`, it would make sense to also rename this method `reset _reference_features`.

Reply:
Updated